src/snovault/elasticsearch/indexer.py

```diff
@@ -293,13 +305,77 @@ def get_current_xmin(request):
     xmin = query.scalar()  # lowest xid that is still in progress
     return xmin
 
+
 class Indexer(object):
     def __init__(self, registry):
         self.es = registry[ELASTIC_SEARCH]
         self.esstorage = registry[STORAGE]
         self.index = registry.settings['snovault.elasticsearch.index']
+        self.queue_server = None
+        self.queue_client = None
+        queue_type = registry.settings.get('queue_type', None)
+        is_queue_server = asbool(registry.settings.get('queue_server'))
+        is_queue_client = asbool(registry.settings.get('queue_client'))
+        queue_client_processes = int(
+            registry.settings.get('queue_client_processes', 1)
+
+        )
+        queue_client_chunk_size = int(
+            registry.settings.get('queue_client_chunk_size', 1024)
+        )
+        queue_client_batch_size = int(
+            registry.settings.get('queue_client_batch_size', 1024)
+        )
+        if is_queue_server:
+            if not queue_type or queue_type == 'Simple':
+                self.queue_server = SimpleUuidServer()
+                if is_queue_client:
```

Comment:
All queue_clients are queue_servers but not the other way around?

Reply:
At some point I had the queue client inheriting from the queue server...or the other way around.  But then I realized that we'll need the queue server to optionally run it's own indexing.   I accomplished this by giving the queue server a client.  Then the serve_queue function the QueueServer checks if it has a client and then will run the worker process with it.

I'm not thrilled about it.  Kinda hacked it in late in the game.